client/wc-api/products/selectors.js

```diff
@@ -0,0 +1,50 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getResourceName } from '../utils';
+import { DEFAULT_REQUIREMENT } from '../constants';
+
+const getProducts = ( getResource, requireResource ) => (
+	query = {},
+	requirement = DEFAULT_REQUIREMENT
+) => {
+	const resourceName = getResourceName( 'product-query', query );
+	const ids = requireResource( requirement, resourceName ).data || [];
+	const products = ids.reduce(
+		( acc, id ) => ( {
+			...acc,
```

Comment:
I _think_ iteratively adding to an array using `.push()` will perform better here. See: https://jsperf.com/es6-add-element-to-create-new-array-concat-vs-spread-op/7

Reply:
I cleaned this up in https://github.com/woocommerce/wc-admin/pull/1391/commits/275356e15585f88dc9f21d0a676dc6e111633c0d to match other selectors.